In [18]:
# Ejemplo de como dividir los grupos de datos.
# Importar librerías 

import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.ensemble import RandomForestRegressor
from sklearn.metrics import r2_score

In [19]:
# Ruta del dataset
file_path = "C:/Git/vehicles_env/datasets/dataset_facebook_cosmetics_us.csv"

# Cargar dataframe
fb = pd.read_csv(file_path, sep=';')

In [20]:
# Crear el dataframe con las características (matriz X)
X = fb.drop(columns='Total Interactions')

# Crear el dataframe con la variable objetivo
y = fb['Total Interactions']

In [21]:
# Dividir los "datos de entrenamiento" y "datos de prueba"
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size= 0.2, random_state=0)

# El parámetro test_size = 0.2 está indicandole que el 80% de los datos serán usados para entrenar
# y el 20% para "datos de prueba"

# Definir el algoritmo del modelo
model = RandomForestRegressor(random_state=0)

# Entrenar el modelo usando "datos de entrenamiento"
model.fit(X_train,y_train)

# Utilizar el modelo para hacer predicciones para los datos de prueba
predictions = model.predict(X_test)

# Estimar R cuadrado utilizando los datos de PRUEBA e imprime el resultado
r2 = r2_score(y_test, predictions)  # escribe tu código aquí
print('Valor R cuadrado: ', r2)

Valor R cuadrado:  0.6702623920180311


¡Es correcto!

El resultado es 0.67. Con un modelo perfecto, este parámetro sería 1, y con modelos absolutamente horribles sería 0. Tu modelo parece decente. Serás capaz de averiguarlo un poco más tarde cuando hablemos en detalle sobre R cuadrado en el siguiente capítulo.